### PR TITLE
docs: Update dev instructions and make targets

### PR DESCRIPTION
**Pull request by patrick-east**
This updates the Makefile targets to remove the hugo-only options for
development, as well as clarifying/simplifying the requirements for
a dev environment. There are now really two options: just look at the
markdown preview with your editor/ide, or install the tools locally
and deploy the full site.

We can likely improve the dev experience some more (would be nice to
get back to the live content updating dev experience) but for now this
should help remove some confusion about how to build and run the site.

Signed-off-by: Patrick East <east.patrick@gmail.com>

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -37,110 +37,31 @@ which are defined in [website/layouts/index.redirects](./webiste/layouts/index.r
...
+changes in this directory are pushed to `master`, the site will be re-built and
+re-deployed.
 
-#### Modifying `content` (*.md)
+## How to Edit and Test
+### Preview Markdown `content` (*.md)
 
-The majorify of this can be done with any markdown renderer (typically built-in or
-a plug-in for IDE's). The rendered output will be very similar to what Hugo will
+The majorify of this can be done with any markdown renderer (typically built into or
```
Comment (tsandall) [inline docs/README.md]: s/majorify/majority/ -- though I kind of like "majorify"

Comment (patrick-east) [reply docs/README.md]: Fixed 😆

**Comment by patrick-east** [comment]
>  I noticed something weird when I tried to use the netlify remote option. The edge pages are empty: 

Huh.. it works for me 😅 https://5e7bdf7ce40fb69353134a87--serene-meitner-49f0a3.netlify.com/docs/edge/

Looking at the content it seems like hugo maybe generated it incorrectly or something.. the content is there we just can't see it

image

What version of hugo did you use? I've got `v0.55.2/extended darwin/amd64`

**Comment by tsandall** [comment]
```
$ hugo version
Hugo Static Site Generator v0.68.3/extended darwin/amd64 BuildDate: unknown
```

Another odd thing I noticed...

When I run `make docs-serve-remote` in the root of the repo on a temp branch with these changes, I see this at the end:

```
# must be run from root of repo for
# the netlify.toml config to work
cd /Users/torin/go/src/github.com/open-policy-agent/opa/docs/.. && netlify deploy
This folder isn't linked to a site yet
? What would you like to do? Link this directory to an existing site

netlify link will connect this folder to a site on Netlify

? How do you want to link this folder to a site? Use current git remote origin (https://github.com/open-policy-agent/opa
)

Looking for sites connected to 'https://github.com/open-policy-agent/opa'...


Directory Linked

Admin url: https://app.netlify.com/sites/openpolicyagent
Site url:  https://www.openpolicyagent.org

Site id saved to /Users/torin/go/src/github.com/open-policy-agent/opa/.netlify/state.json

You can now run other `netlify` cli commands in this directory
Deploy path:        /Users/torin/go/src/github.com/open-policy-agent/opa/docs/website/public
Configuration path: /Users/torin/go/src/github.com/open-policy-agent/opa/netlify.toml
Deploying to draft URL...
✔ Finished hashing 573 files
✔ CDN requesting 1 files
✔ Finished uploading 1 assets
✔ Draft deploy is live!

Logs:           https://app.netlify.com/sites/openpolicyagent/deploys/5e7cbcf7c61f678330c00658
Live Draft URL: https://5e7cbcf7c61f678330c00658--openpolicyagent.netlify.com

If everything looks good on your draft URL, take it live with the --prod flag.
netlify deploy --prod

```

This is a little bit odd:

```

? How do you want to link this folder to a site? Use current git remote origin (https://github.com/open-policy-agent/opa
)


Looking for sites connected to 'https://github.com/open-policy-agent/opa'...
```

Given my origin remote is my fork:

```
$ git remote -v | grep origin
origin	git@github.com:tsandall/opa.git (fetch)
origin	git@github.com:tsandall/opa.git (push)
```

Anyway...not a major issue for me.

**Pull request merged**